internal/js/modules/k6/browser/common/page.go

```diff
@@ -485,6 +488,32 @@ func (p *Page) urlTagName(url string, method string) (string, bool) {
 	return newTagName, urlMatched
 }
 
+func (p *Page) onRequest(request *Request) {
+	if !hasPageOnHandler(p, EventPageRequestCalled) {
+		return
+	}
+
+	p.eventHandlersMu.RLock()
+	defer p.eventHandlersMu.RUnlock()
+	for _, h := range p.eventHandlers[EventPageRequestCalled] {
+		err := func() error {
+			// Handlers can register other handlers, so we need to
+			// unlock the mutex before calling the next handler.
+			p.eventHandlersMu.RUnlock()
+			defer p.eventHandlersMu.RLock()
```

Comment:
Why do we need to lock again? Isn't the lock before already sage-guarding for it?

Reply:
Yeah, this looks a bit odd. The reason behind this is to allow the handler to be able to add more handlers. This is a behaviour that Playwright exhibits and also documents, so it's something we're replicating.

```js
page.on('request', async () => {
    page.on('response', async () => {
        // Do something with the request and response data...
    })
})
```

Reply:
@codebien I asked the same question before for another related `page.on` method. Here's a little bit more detail @ankur22 previously answered: https://github.com/grafana/xk6-browser/pull/1456#discussion_r1786305452